xilem_web/src/after_update.rs

```diff
@@ -1,55 +1,107 @@
 // Copyright 2023 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-use xilem_core::{MessageResult, Mut, View, ViewId, ViewPathTracker};
+use xilem_core::{MessageResult, Mut, View, ViewId, ViewMarker};
 
-use crate::{interfaces::Element, DynMessage, ViewCtx};
+use crate::{DomView, DynMessage, ViewCtx};
 
-pub struct AfterUpdate<E, F> {
-    pub(crate) element: E,
-    pub(crate) callback: F,
+pub struct AfterBuild<E, F> {
+    element: E,
+    callback: F,
 }
 
-impl<E, F> AfterUpdate<E, F> {
-    pub fn new(element: E, callback: F) -> AfterUpdate<E, F> {
-        AfterUpdate { element, callback }
-    }
+pub struct AfterRebuild<E, F> {
+    element: E,
+    callback: F,
 }
 
-pub struct AfterUpdateState<E, S> {
+pub struct BeforeTeardown<E, F> {
     element: E,
-    child_state: S,
-    child_id: ViewId,
+    callback: F,
+}
+
+impl<E, F> AfterBuild<E, F> {
+    pub fn new(element: E, callback: F) -> AfterBuild<E, F> {
+        Self { element, callback }
+    }
+}
+
+impl<E, F> AfterRebuild<E, F> {
+    pub fn new(element: E, callback: F) -> AfterRebuild<E, F> {
+        Self { element, callback }
+    }
+}
+
+impl<E, F> BeforeTeardown<E, F> {
+    pub fn new(element: E, callback: F) -> BeforeTeardown<E, F> {
+        Self { element, callback }
+    }
+}
+
+impl<E, F> ViewMarker for AfterBuild<E, F> {}
+impl<E, F> ViewMarker for AfterRebuild<E, F> {}
+impl<E, F> ViewMarker for BeforeTeardown<E, F> {}
+
+impl<State, V, F> View<State, (), ViewCtx, DynMessage> for AfterBuild<V, F>
+where
+    State: 'static,
+    F: Fn(&V::DomNode) + 'static,
+    V: DomView<State> + 'static,
+{
+    type Element = V::Element;
+
+    type ViewState = V::ViewState;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let (el, view_state) = self.element.build(ctx);
+        (self.callback)(&el.node);
```

Comment:
I think this is fine for now, as it doesn't break the modifier system (and the issue explained here is less worrisome), but as explained earlier, this should have the props applied to the element before calling this IMO.
Probably worth adding a `TODO` comment.